debezium-server-eventhubs/src/main/java/io/debezium/server/eventhubs/BatchManager.java

```diff
@@ -138,6 +144,10 @@ public void sendEventToPartitionId(EventData eventData, Integer recordIndex, Int
     public void sendEventWithDynamicPartitionKey(EventData eventData, String partitionKey) {
         String effectivePartitionKey = Strings.isNullOrBlank(partitionKey) ? "default" : partitionKey;
 
+        if (hashMessageKey) {
+            effectivePartitionKey = applyHashFunction(effectivePartitionKey, hashMessageFunction);
```

Comment:
IMHO it makes sense to move hash calculation into `EventHubsChangeConsumer` and pass either hashed or not hashed value and `BatchManager` will be unchanged.